libsel4vm/src/guest_ram.c

```diff
@@ -48,7 +48,14 @@ static int ram_region_cmp(const void *a, const void *b)
 {
     const vm_ram_region_t *aa = a;
     const vm_ram_region_t *bb = b;
-    return aa->start - bb->start;
+
+    if (aa->start == bb->start) {
+        return 0;
+    } else if (aa->start < bb->start) {
+        return -1;
+    } else {
+        return 1;
+    }
```

Comment:
The problem here is that the function returns an `int` while `start` is an `uintptr_t` and the diff between aa and bb may be bigger than 32-bits. So the fix is fine, but the commit message doesn't make much sense to me.

Reply:
Updated the commit message to include that information and add that this function is used with qsort.

Reply:
Typo: "to sort to guest ram regions." -> "to sort the guest ram regions.", otherwise great. Good to mention qsort.

Reply:
Thanks. But (nitpicking) I think it would be better to put the qsort usage as comment over the function and not in the commit message then.

Reply:
I added a comment about qsort return values.